server/memphis_handlers_tags.go

```diff
@@ -0,0 +1,467 @@
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+}
+
+func DeleteTagsBySchema(name string) {
+	exist, schema, err := IsSchemaExist(name)
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+	if !exist {
+		serv.Errorf("Schema with this name does not exist")
+		return
+	}
+	_, err = tagsCollection.UpdateMany(context.TODO(), bson.M{}, bson.M{"$pull": bson.M{"schemas": schema.ID}})
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+}
+
+func DeleteTagsByUser(name string) {
+	exist, user, err := IsUserExist(name)
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+	if !exist {
+		serv.Errorf("User with this name does not exist")
+		return
+	}
+	_, err = tagsCollection.UpdateMany(context.TODO(), bson.M{}, bson.M{"$pull": bson.M{"users": user.ID}})
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+}
+
+func checkIfEmptyAndDelete(name string) {
+	exist, tag, _ := IsTagExist(name)
+	if exist {
+
+		if len(tag.Schemas) == 0 && len(tag.Stations) == 0 && len(tag.Users) == 0 {
+			_, err := tagsCollection.DeleteOne(context.TODO(), bson.M{"_id": tag.ID})
+			if err != nil {
+				serv.Warnf("Delete tag error:" + err.Error())
+			}
+		}
+	}
+}
+
+func (th TagsHandler) CreateTags(c *gin.Context) {
+	var body models.CreateTagSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	for _, tag := range body.Tags {
+		name := strings.ToLower(tag.Name)
+		err := AddTagToEntity(name, body.EntityType, body.EntityName, tag.ColorBG, tag.ColorTXT)
+		if err != nil {
+			serv.Errorf("Failed creating tag: %v", err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+	}
+	c.IndentedJSON(200, []string{})
+}
+
+func (th TagsHandler) RemoveTags(c *gin.Context) {
+	var body models.RemoveTagsSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	for _, tagName := range body.Names {
+		name := strings.ToLower(tagName)
+		exist, tag, err := IsTagExist(name)
+
+		if err != nil {
+			serv.Errorf("RemoveTags error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		if !exist {
+			c.IndentedJSON(200, []string{})
+			return
+		}
+		switch body.EntityType {
+		case "station":
+			station_name, err := StationNameFromStr(body.EntityName)
+			if err != nil {
+				serv.Errorf("RemoveTags error: " + err.Error())
+				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+				return
+			}
+			exist, station, err := IsStationExist(station_name)
+			if err != nil {
+				serv.Errorf("RemoveTags error: " + err.Error())
+				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+				return
+			}
+			if !exist {
+				serv.Errorf("Station with this name does not exist")
+				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
```

Comment:
this is not a server error, return the user the proper error with status code 666